components/popconfirm/popconfirm.ts

```diff
@@ -268,8 +274,28 @@ export class NzPopconfirmComponent extends NzToolTipComponent implements OnDestr
   }
 
   onConfirm(): void {
-    this.nzOnConfirm.next();
-    super.hide();
+    if (this.nzBeforeConfirm) {
+      const observable = wrapIntoObservable(this.nzBeforeConfirm()).pipe(first());
+      this.confirmLoading = true;
+      observable
+        .pipe(
+          finalize(() => {
+            this.confirmLoading = false;
+            this.cdr.markForCheck();
+          }),
+          takeUntil(this.nzVisibleChange),
+          takeUntil(this.destroy$)
+        )
+        .subscribe(value => {
+          if (value) {
+            this.nzOnConfirm.next();
```

Comment:
L291-292 & L296-297 could be extracted to a function.

Reply:
updated！